nemo/utils/app_state.py

```diff
@@ -22,6 +22,7 @@ def __init__(self):
         self._app_cfg = None
 
         # World info
+        self._device_id = None
         self._local_rank = None
```

Comment:
Is device id semantically different from local_rank?

Reply:
Device id is the CUDA device id. Local rank is assigned by the distributed system. Often they are the same.

Reply:
Ah ok.